Services/ProductsService.js

```diff
@@ -1,14 +1,31 @@
 const getAllProducts = async (connection) => {
     return await connection.query("SELECT `name`, `price`, `SKU` FROM `products`;")
-
 }
 
 const addSingleProduct = async(connection, sku, name, price, stock_level) => {
     return await connection.query('INSERT `products` (`SKU`, `name`, `price`,`stock_level`) VALUES' + `('${sku}', '${name}', '${price}', '${stock_level}');`)
 }
 
+const getSingleProduct = async (connection, SKU) => {
+    const verifiedSKU = async (SKU) => {
+        const regex = /[A-Za-z]{3}-[0-9A-Za-z]{1,3}-[A-Za-z]{3}/
+        const specChar = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/
+        const normaliseString = SKU.toUpperCase()
+        let response
+        if (normaliseString.length > 11 || normaliseString.length < 9 && specChar.test(normaliseString)){
+            response = ''
+        } else {
+            regex.test(SKU)
+                ? response = await connection.query('SELECT `SKU`, `name`, `price`, `stock_level` FROM `products` WHERE `SKU` =' + `'${normaliseString}'`)
```

Comment:
I would add 'LIMIT 1' to this conneciton.query
full sql query would be equivalent of this in SQL 'SELECT `SKU`, `name`, `price`, `stock_level` FROM `products` WHERE `SKU` = 'CRO-3-GRE' LIMIT 1;'
This means that, even when there are duplicates in the database, only 1 item will be returned

Reply:
Ye there is no harm in doing that, I merely didn't as there shouldn't be duplicates.....should we be checking for this when adding an item?

Reply:
Amended